src/shared/views/RecommendedCards.js

```diff
@@ -77,13 +78,92 @@ type OrderedProject = {
 
 type Props = {
     announcement: Object,
+    firebase: Object,
     navigation: NavigationProp,
     projects: Array<OrderedProject>,
 };
 
+// request only active projects from firebase (status === 'active')
+// firebase doesn't allow multiple query params, so for project types we filter in render()
+// but here we can still limit to 20 projects maximum
+// `path` defines where the resulting data is copied in the redux store
+// (state.firebase.ordered.projects in this case, because we've asked for `orderByChild`)
+const projectsQuery = {
+    isQuery: true,
+    path: 'v2/projects',
+    queryId: 'projectsQuery',
+    queryParams: ['orderByChild=status', 'equalTo=active', 'limitToFirst=20'],
+    storeAs: 'projects',
+    type: 'value',
+};
+// load any announcement data from firebase
+// (state.firebase.data.announcement here because we've not ordered the query)
+const announcementQuery = {
+    isQuery: true,
+    path: 'v2/announcement',
+    queryId: 'announcementQuery',
+    queryParams: ['limitToLast=2'],
+    storeAs: 'announcement',
+    type: 'value',
+};
+
 class _RecommendedCards extends React.Component<Props> {
     tutorialModal: ?Modal;
 
+    willBlurAnnouncementSubscription: NavigationEventSubscription;
+
+    willFocusAnnouncementSubscription: NavigationEventSubscription;
+
+    willBlurProjectSubscription: NavigationEventSubscription;
+
+    willFocusProjectSubscription: NavigationEventSubscription;
+
+    componentDidMount() {
+        this.subscribeToProjects();
+        this.subscribeToAnnouncements();
+    }
+
+    componentWillUnmount() {
+        this.willFocusAnnouncementSubscription.remove();
+        this.willBlurAnnouncementSubscription.remove();
+        this.willFocusProjectSubscription.remove();
+        this.willBlurProjectSubscription.remove();
+    }
+
+    subscribeToAnnouncements = () => {
+        const { type, path, storeAs, ...options } = announcementQuery;
+        const { firebase, navigation } = this.props;
+        this.willFocusAnnouncementSubscription = navigation.addListener(
```

Comment:
What are `willFocus` and `willBlur` refering to? 

Is it like this?
*  users opens the project view, this "connects" to firebase projects and the screen is updated live
* when user selects a project and starts mapping, this triggers `willBlur` and the projects are disconnected from firebase

Reply:
`willFocus` is triggered when the project list is about to be shown, and we then connect to firebase to get live updates on each change to the query we've sent. This continues until we `unWatch` it, which happens when the project list screen loses foces, in `willBlur`. Then we stop listening to firebase, so whatever happens on the db will not impact us.